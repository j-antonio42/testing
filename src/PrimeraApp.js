import React from 'react'

const PrimeraApp = ({saludo, subtitulo = 'soy el subtitulo lol'}) => {
    return (
        <>
        <h1>{ saludo }</h1>
        <h2>{ subtitulo }</h2>   
        </>
    )
}

export default PrimeraApp

