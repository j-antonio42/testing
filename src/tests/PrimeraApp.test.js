import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {

   /* test('debe mostrar el mensaje', () => {

        const saludo = 'Hola mundo'

        const wrapper = render()
        
    }); */

    test('debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola te pico la cola'
        const wrapper = shallow( <PrimeraApp saludo = {saludo} /> )

        expect( wrapper ).toMatchSnapshot()
        
    });

    test('debe mostrar subtitulo enviado por props', () => {

        const saludo = 'Hola te pico la cola'
        const subtitulo = 'soy el subtitulo lol'

        const wrapper = shallow( 
            <PrimeraApp 
                saludo = {saludo} 
                subtitulo = {subtitulo}
            /> )

        
        
    });
    
    
});
