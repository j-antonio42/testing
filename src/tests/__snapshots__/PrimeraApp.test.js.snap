// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola te pico la cola
  </h1>
  <h2>
    soy el subtitulo lol
  </h2>
</Fragment>
`;
